app/brave_settings_strings.grdp

```diff
@@ -302,6 +299,9 @@
   <message name="IDS_SETTINGS_BRAVE_SHIELDS_DEFAULTS_ADVANCED_VIEW_LABEL" desc="Default Brave advanced view title label">
     Advanced view
   </message>
+  <message name="IDS_SETTINGS_FORGET_FIRST_PARTY_STORAGE" desc="Toggle label">
+    Forget me when I close a site
```

Comment:
@rmcfadden3 I think `Forget me when I close the site` [below](https://github.com/brave/brave-core/pull/18014/files?show-viewed-files=true&file-filters%5B%5D=#diff-f2536e2302c1f665f39278809f9413259fbcfcf5177441fbb21d3bcb51b48404R113) in `brave_shields_strings.grdp` in the context of the Shields panel is clear enough. But, do you think `Forget me when I close a site` in the context of the Shields settings on the settings page would be descriptive enough? Could this toggle benefit from having an explanation text added?

Reply:
@mkarolin — can you summarize here exactly what happens when this toggle is switched to "On" position for a site, and you then navigate away from that site? Like, does it clear all cookies for that site, and nuke it from your browser history?

Reason I ask is I think some people might already assume Brave does this, so we may need a slightly more descriptive text.

And, yes, to answer your question: I agree that adding explanation text would help.

Reply:
I think @goodov would be able to answer those questions better.

Reply:
@rmcfadden3 
The features clears cookies and any storage for the site (local storage, indexed db) after the website is closed and a short keep-alive period (30 seconds), but does not remove it from the history.

Reply:
ping @rmcfadden3 @mkarolin 
what's the decision? I can use `Forget me when I close the site` in both places for now (to not mess with a/the article in the shields popup and the settings page), but please tell me if it's okay.

Reply:
@pes10k @goodov @mkarolin — yeah, I suppose saying forget "me" is clear enough, in that it clears personal info, but not browsing history.

In the context of Shields settings on the settings page, I do think it would be helpful to include explanatory text. I'm open to your edits, but something like:

**`Forget me when I close this site`**
`Clears cookies, local storage, etc when you close a site.`

Reply:
@rmcfadden3 @pes10k 

Made it similar to "Clear browsing data" window:
![image](https://user-images.githubusercontent.com/5928869/234275951-3f7e5c10-dc70-4532-abbb-ec3830e09541.png)